### PR TITLE
Closes #441 | Add dataloader for ASR-SMALDUSC

### LINKED ISSUE #441: Create dataset loader for ASR-SMALDUSC: A SCRIPTED MALAY DAILY-USE SPEECH CORPUS
Dataloader name: `asr_smaldusc/asr_smaldusc.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?asr_smaldusc

|     Dataset| asr_smaldusc |
|-------------|---|
| Description | This open-source dataset consists of 4.8 hours of transcribed Malay scripted speech focusing on daily use sentences, where 2,839 utterances contributed by ten speakers were contained. |
| Subsets | - |
| Languages | zlm |
| Tasks | Automatic Speech Recognition, Text-To-Speech Synthesis |
| License | Creative Commons Attribution Non Commercial No Derivatives 4.0 (cc-by-nc-nd-4.0) |
| Homepage | https://magichub.com/datasets/malay-scripted-speech-corpus-daily-use-sentence/ |
| HF URL | - |
| Paper URL | - |

### PR DESCRIPTION
**Pull request by zwenyu**
Closes #441 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/{my_dataset}/{my_dataset}.py` (please use only lowercase and underscore for dataset folder naming, as mentioned in dataset issue) and its `__init__.py` within `{my_dataset}` folder.
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_LOCAL`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py` or `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py --subset_id {subset_name_without_source_or_seacrowd_suffix}`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Review by MJonibek — APPROVED** [review]
@zwenyu Thank you for working on this dataloader. Checked, works good, LGTM

File: seacrowd/sea_datasets/asr_smaldusc/asr_smaldusc.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                    "split": "train",
+                },
+            )
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+
+        # read UTTRANSINFO file
+        # columns: channel, uttrans_id, speaker_id, prompt (empty field), transcription
```
Comment (akhdanfadh) [inline seacrowd/sea_datasets/asr_smaldusc/asr_smaldusc.py:127]: Any reason why do you omit the `prompt` field for the `source` schema? It's not empty and can be accessed based on your code using `uttransinfo_data[i][3]`.

**Review by akhdanfadh — COMMENTED** [review]
Tested and worked, only one question from me.

File: seacrowd/sea_datasets/asr_smaldusc/asr_smaldusc.py (hunk 1 above)
Comment (zwenyu) [reply seacrowd/sea_datasets/asr_smaldusc/asr_smaldusc.py:127]: I meant the prompts are empty strings

Comment (akhdanfadh) [reply seacrowd/sea_datasets/asr_smaldusc/asr_smaldusc.py:127]: It's not? This is what I got running your code.

```python
>>> import os
>>> from pathlib import Path
>>> from datasets import DownloadManager
>>> URL = "https://magichub.com/df/df.php?file_name=Malay_Scripted_Speech_Corpus_Daily_Use_Sentence.zip"
>>> filepath = Path(DownloadManager().download_and_extract(URL))
>>> uttransinfo_filepath = os.path.join(filepath, "UTTRANSINFO.txt")
>>> with open(uttransinfo_filepath, "r", encoding="utf-8") as uttransinfo_file:
...     uttransinfo_data = uttransinfo_file.readlines()
... 
>>> uttransinfo_data = uttransinfo_data[1:]  # remove header
>>> uttransinfo_data = [s.strip("\n").split("\t") for s in uttransinfo_data]
>>> uttransinfo_data[:5]
[['C1', 'G5001_1_S0002.wav', 'G5001', 'Dia tidak tahu apa-apa .', 'dia tidak tahu apa-apa'], ['C1', 'G5001_1_S0003.wav', 'G5001', 'Dia segera beredar dari situ .', 'dia segera beredar dari situ'], ['C1', 'G5001_1_S0005.wav', 'G5001', 'Dia menoleh ke arah gadis itu .', 'dia menoleh ke arah gadis itu'], ['C1', 'G5001_1_S0006.wav', 'G5001', 'Lelaki itu memandangnya dari atas ke bawah .', 'lelaki itu memandang dari atas ke bawah'], ['C1', 'G5001_1_S0007.wav', 'G5001', 'Dia menarik nafas dalam-dalam dan menghembuskannya dengan perlahan .', 'dia menarik nafas dalam-dalam dan menghembuskannya dengan perlahan']]
```

Comment (zwenyu) [reply seacrowd/sea_datasets/asr_smaldusc/asr_smaldusc.py]: I checked again and yes you are right, it does have the prompt. I was looking at another similar dataset. I've added the prompt field and pushed the change. Thanks for catching.

**Review by akhdanfadh — COMMENTED** [review]
LGTM, tested and worked. Thank you for the dataloader @zwenyu! Merging this.

**Pull request merged**